### DISCUSSION TITLE
Access to HttpProxy.Timeout like DefaultUrl

### DISCUSSION THREAD

**Discussion by bdeluard** [MarimerLLC/csla#2227] [Questions]
Hi,

I need to change the http timeout for the dataportal calls from wpf clients.

I can't change the Timeout property of HttpProxy class and this property is not initialized from a default static value (initialized from  an ApplicationContext default property.

I haver the idea to subclass HttpProxy to add this code in a MyHttpProxy class, but I don't find where to replace the use of HttpProxy by MyHttpProxy.

How could I do that?

Thanks in advance,

Bruno

**Comment by rockfordlhotka** [comment]
You are correct to create a subclass.

As your app starts up, you set the type of proxy for the data portal. In WPF you might be doing this in code or through app.config. Either way, use your custom type name instead of `Csla.DataPortalClient.HttpProxy`.

**Reply by bdeluard** [reply]
Ok thanks for the reply.

But where I instruct this fact to use my class instead of the base class?

Thanks!

**Reply by sylum** [reply]
Hi bdeluard,

Below, a quick sample to how subclass HttpProxy class

```
 // https://github.com/MarimerLLC/csla/blob/main/Source/Csla.Web.Mvc.Shared/Server/Hosts/HttpPortalController.cs
    public class CustomHttpProxy : Csla.DataPortalClient.HttpProxy
    {
        private static HttpClient _client;

        public CustomHttpProxy()
        {
            Timeout = (1000 * 300); // Set your timeout
        }

        protected override HttpClient GetHttpClient()
        {
            try
            {
                if (_client == null)
                {
                    _client = new HttpClient();

                    if (this.Timeout > 0)
                    {
                        _client.Timeout = TimeSpan.FromMilliseconds(this.Timeout);
                    }
                }
            }
            catch (Exception ex)
            {
                throw ex;
            }
            return _client;
        }
    }
```

Next, you can use it, on the csla configuration call.
```
 CslaConfiguration.Configure()
                  .DataPortal()
                  .DefaultProxy(typeof(CustomHttpProxy),
                   ConfigSingleton.Instance.CslaDataPortalUrlSettings);
```

**Reply by bdeluard** [reply]
Very nice!

Great thanks,

Bruno
